bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/FileDialogResult.java

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.swt.widgets;
+
+/**
+ * A representation of the result of file dialogs such as the {@link FileDialog}
+ * or {@link DirectoryDialog}.
+ *
+ * @since 3.125
+ */
+public class FileDialogResult {
+	public static enum State {
+		/**
+		 * The user has successfully selected a file or folder
+		 */
+		SELECTED,
+		/**
+		 * The user has canceled the dialog
+		 */
+		CANCELED,
+		/**
+		 * An unspecified error occurred while the dialog was open
+		 */
+		UNKNOWN_ERROR
+	}
+
+	private String path;
+
+	private State state;
+
+	private FileDialogResult(String path, State state) {
+		this.path = path;
+		this.state = state;
+	}
+
+	/**
+	 * {@return a string describing the absolute path of the selected file or folder
+	 * or <code>null</code> if the dialog was canceled or if there was an error}
+	 */
+	public String getSelectedPath() {
+		return path;
+	}
+
+	/**
+	 * {@return the result state of the dialog, indicating a successful selection,
+	 * user cancellation and other errors}
+	 */
+	public State getState() {
+		return state;
+	}
+
+	static FileDialogResult createForSuccessfulSelection(String path) {
+		return new FileDialogResult(path, State.SELECTED);
+	}
+
+	static FileDialogResult createForUserCancellation() {
+		return new FileDialogResult(null, State.CANCELED);
+	}
+
+	static FileDialogResult createForUnkownError() {
```

Comment:
```suggestion
	static FileDialogResult createForError(SWTException exception) {
```

I would simply add a `SWTException` here and allow to query that, implementations can then pass additional details in the exception, it could be printed with a stacktrace to track the location of it and so on

Reply:
The main point for me is that "error" does not mean anything, what should I show to the user? And why is not the dialog already handle this (e.g. does not allow me to select a patch that is too long?).

So this case seems really "exceptional" for me and I don't want to have the boilerplate to check each and every call if it might be of an unknown error, stead I assume when the method returns everything is normale so either:

1. I get a (valid) path
2. I get nothing (user canceled)
3. I get an Exception (something went wrong)

e.g. "NO MORE HANDLES" or a like it also not something by SWT so SWTExceptions (for me) are more a way to signal an native error one can't forsee or handle in client code (like here).